support/android/apk/servoapp/src/main/java/org/mozilla/servo/MainActivity.java

```diff
@@ -255,4 +255,15 @@ public void onMediaSessionPlaybackStateChange(int state) {
           return;
       }
     }
+
+    @Override
+    public void onMediaSessionSetPositionState(float duration, float position, float playbackRate) {
+        Log.d("onMediaSessionSetPositionState", duration + " " + position + " " + playbackRate);
+        if (mMediaSession == null) {
+            mMediaSession = new MediaSession(mServoView, this, getApplicationContext());
+        }
+
+        mMediaSession.setPositionState(duration, position, playbackRate);
```

Comment:
I have a wrong implementation.
Should I call `onMediaSessionSetPositionState` of `Servo.java` directly here?
Or is it better to implement `setPositionState ` in `MediaSession.java` and call it there?

Reply:
`Servo.onMediaSessionSetPositionState` is what calls this method (note that `MainActivity` implements the `Servo.Client` interface).

We need to implement `MediaSession.setPositionState`. That's where we should modify the existing notification UI to add the current position, the total duration and a progress bar. If you don't feel like hacking on the Android UI, leave the method unimplemented.

Reply:
Thanks!
I don't have a good understanding of the Android UI.
So this time I will not implement it. Sorry.